### ISSUE TITLE
C++ side of Ale build for 0.8.0 linked to Framework Python for MacOS

### ISSUE THREAD

**Issue by Kelvinrr** [conda-forge/ale-feedstock#21]
Issue:

Seems like the MacOS 0.8.0 C++ library is linked against the Python Framework normally located at `/Library/Frameworks/Python.framework/` instead of the conda release of python. This is causing build errors in ISIS (and presumably anyone else who trying using the C++ library on MacOS). 

otool -L output: 
```
(isis) igswzawglt0091:isis kelvin$ otool -L /Users/kelvin/anaconda3/envs/isis/lib/libale.0.dylib
/Users/kelvin/anaconda3/envs/isis/lib/libale.0.dylib:
	@rpath/libale.0.dylib (compatibility version 0.0.0, current version 0.8.0)
	@rpath/Python3.framework/Versions/3.7/Python3 (compatibility version 3.7.0, current version 3.7.0)
	@rpath/libc++.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
```

compared to 0.7.2: 

```
(aleisis) igswzawglt0091:ISIS3 kelvin$ otool -L /Users/kelvin/anaconda3/envs/aleisis/lib/libale.dylib
/Users/kelvin/anaconda3/envs/aleisis/lib/libale.dylib:
	@rpath/libale.1.dylib (compatibility version 1.0.0, current version 0.3.0)
	@rpath/libgsl.23.dylib (compatibility version 25.0.0, current version 25.0.0)
	@rpath/libcblas.3.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libpython3.7m.dylib (compatibility version 3.7.0, current version 3.7.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.50.4)
```

see also: https://astroservices.usgs.gov/jenkins/job/ISIS-Builds/job/dev/134/execution/node/94/log/ for logs of a linking failure. 

<br/>
Environment (<code>conda list</code>):
<details>

```
(isis) igswzawglt0091:isis kelvin$ conda list
# packages in environment at /Users/kelvin/anaconda3/envs/isis:
#
# Name                    Version                   Build  Channel
ale                       0.8.0            py37ha1cc60f_0    conda-forge
armadillo                 9.200.4         blas_openblash45cfd7e_202  [blas_openblas]  conda-forge
arpack                    3.5.0           blas_openblas_202  [blas_openblas]  conda-forge
blas                      1.1                    openblas    conda-forge
boost                     1.72.0           py37hbf1eeb5_0    conda-forge
boost-cpp                 1.72.0               hdf9ef73_0    conda-forge
bullet                    2.86.1                        0    usgs-astrogeology
bzip2                     1.0.8                h0b31af3_2    conda-forge
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
cairo                     1.16.0            hec6a9b0_1003    conda-forge
certifi                   2020.4.5.1       py37hc8dfbb8_0    conda-forge
cfitsio                   3.470                h389770f_2    conda-forge
cmake                     3.17.0               hd28f656_0    conda-forge
cspice                    66                h0b31af3_1007    conda-forge
curl                      7.68.0               h8754def_0    conda-forge
decorator                 4.4.2                      py_0    conda-forge
eigen                     3.3.7             ha1b3eb9_1001    conda-forge
embree                    2.16.0               h6834224_0    usgs-astrogeology
expat                     2.2.9                h4a8c4bd_2    conda-forge
ffmpeg                    4.1.3                h5c2b479_0    conda-forge
flann                     1.9.1             h3c234d5_1007    conda-forge
fontconfig                2.13.1            h6b1039f_1001    conda-forge
freetype                  2.10.1               h8da9a1a_0    conda-forge
freexl                    1.0.5             h1de35cc_1002    conda-forge
gdal                      3.0.2            py37h97c3584_4    conda-forge
geos                      3.7.2                h6de7cb9_2    conda-forge
geotiff                   1.5.1                ha113b03_7    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
giflib                    5.1.7                h01d97ff_1    conda-forge
glib                      2.64.2               h577aef8_0    conda-forge
gmm                       5.0                  h6aef312_0    usgs-astrogeology
gmp                       6.2.0                h4a8c4bd_2    conda-forge
gnutls                    3.6.5             h53004b3_1002    conda-forge
graphite2                 1.3.13            h12caacf_1001    conda-forge
gsl                       2.6                  ha2d443c_0    conda-forge
harfbuzz                  2.4.0                hd8d2a14_3    conda-forge
hdf4                      4.2.13            h84186c3_1003    conda-forge
hdf5                      1.10.5          nompi_h0cbb7df_1103    conda-forge
icu                       64.2                 h6de7cb9_1    conda-forge
jama                      125                           0    usgs-astrogeology
jasper                    1.900.1           h636a363_1006    conda-forge
jpeg                      9c                h1de35cc_1001    conda-forge
json-c                    0.13.1            h1de35cc_1001    conda-forge
kakadu                    1                             0    usgs-astrogeology
kealib                    1.4.13               h2324030_0    conda-forge
krb5                      1.16.4               h1752a42_0    conda-forge
lame                      3.100             h1de35cc_1001    conda-forge
libblas                   3.8.0               11_openblas    conda-forge
libcblas                  3.8.0               11_openblas    conda-forge
libclang                  8.0.1                ha1b3eb9_2    conda-forge
libcurl                   7.68.0               h709d2b2_0    conda-forge
libcxx                    9.0.1                         1    conda-forge
libdap4                   3.20.4               habf5908_0    conda-forge
libedit                   3.1.20170329      hcfe32e1_1001    conda-forge
libffi                    3.2.1             h4a8c4bd_1007    conda-forge
libgcc                    4.8.5               hdbeacc1_10    conda-forge
libgdal                   3.0.2                h59ca9cb_4    conda-forge
libgfortran               3.0.1                         0    conda-forge
libiconv                  1.15              h0b31af3_1006    conda-forge
libkml                    1.3.0             h169b8f9_1011    conda-forge
liblapack                 3.8.0               11_openblas    conda-forge
liblapacke                3.8.0               11_openblas    conda-forge
libllvm8                  8.0.1                h770b8ee_0    conda-forge
libnetcdf                 4.7.1           nompi_hec86efb_102    conda-forge
libopenblas               0.3.6                hdc02c5d_2
libopencv                 3.4.9                    py37_2    conda-forge
libpng                    1.6.37               hbbe82c9_1    conda-forge
libpq                     11.5                 h31a01ba_2    conda-forge
libprotobuf               3.11.4               hd174df1_0    conda-forge
libspatialite             4.3.0a            h7a659be_1032    conda-forge
libssh2                   1.8.2                hcdc9a53_2    conda-forge
libtiff                   4.1.0                ha78913b_3    conda-forge
libuv                     1.34.0               h0b31af3_0    conda-forge
libwebp                   1.0.2                h20df551_3    conda-forge
libxml2                   2.9.10               h53d96d6_0    conda-forge
llvm-openmp               10.0.0               h28b9765_0    conda-forge
lz4-c                     1.8.3             h6de7cb9_1001    conda-forge
make                      4.3                  h0b31af3_0    conda-forge
mesalib                   18.0.0               hb6cfc13_0    conda-forge
metis                     5.1.0             h4a8c4bd_1005    conda-forge
mysql                     5.7.20            h0a44026_1001    conda-forge
mysql-connector-c         6.1.11            had4075f_1006    conda-forge
nanoflann                 1.3.1                         0    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
nettle                    3.4.1             h3efe00b_1002    conda-forge
networkx                  2.4                        py_1    conda-forge
ninja                     1.10.0               ha1b3eb9_0    conda-forge
nlohmann_json             3.7.3                h4a8c4bd_1    conda-forge
nn                        1.86.0            h1de35cc_2003    conda-forge/label/gcc7
nspr                      4.20              h0a44026_1000    conda-forge
nss                       3.47                 hc0980d9_0    conda-forge
numpy                     1.15.2          py37_blas_openblashd3ea46f_1  [blas_openblas]  conda-forge
openblas                  0.2.20                        8    conda-forge
opencv                    3.4.9                    py37_2    conda-forge
openh264                  1.8.0             hd9629dc_1000    conda-forge
openjpeg                  2.3.1                h254dc36_3    conda-forge
openssl                   1.1.1g               h0b31af3_0    conda-forge
pcl                       1.9.1             h3d0f52c_1005    conda-forge
pcre                      8.44                 h4a8c4bd_0    conda-forge
pip                       20.0.2                     py_2    conda-forge
pixman                    0.38.0            h01d97ff_1003    conda-forge
poppler                   0.67.0               h16886b5_8    conda-forge
poppler-data              0.4.9                         1    conda-forge
postgresql                11.5                 h26bc10f_2    conda-forge
proj                      6.2.1                h773a61f_0    conda-forge
protobuf                  3.11.4           py37h570ac47_1    conda-forge
pvl                       0.3.0                      py_1    conda-forge
py-opencv                 3.4.9            py37h5ca1d4c_2    conda-forge
python                    3.7.6           h90870a6_5_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2019.3                     py_0    conda-forge
pyyaml                    5.3.1            py37h9bfed18_0    conda-forge
qhull                     7.2.0                         0    usgs-astrogeology
qt                        5.12.5               h1b46049_0    conda-forge
qwt                       6.1.4             h2815f34_1005    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
rhash                     1.3.6             h1de35cc_1001    conda-forge
scipy                     1.3.1            py37hab3da7d_2    conda-forge
setuptools                46.1.3           py37hc8dfbb8_0    conda-forge
six                       1.14.0                     py_1    conda-forge
spiceypy                  2.3.2                      py_0    conda-forge
sqlite                    3.30.1               h93121df_0    conda-forge
suitesparse               4.5.6           blas_openblash55dcaef_201  [blas_openblas]  conda-forge
superlu                   5.2.1           blas_openblash47f33a9_1203  [blas_openblas]  conda-forge/label/gcc7
tbb                       2018.0.5             h2d50403_0    conda-forge
tiledb                    1.7.0                hd5e958f_2    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
tnt                       126                           0    usgs-astrogeology
tzcode                    2019a             h01d97ff_1002    conda-forge
wheel                     0.34.2                     py_1    conda-forge
x264                      1!152.20180806       h1de35cc_0    conda-forge
xerces-c                  3.2.2             h8f8adb3_1004    conda-forge
xz                        5.2.5                h0b31af3_0    conda-forge
yaml                      0.2.4                h0b31af3_0    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
zstd                      1.4.4                hed8d7c8_2    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
(isis) igswzawglt0091:isis kelvin$ conda info

     active environment : isis
    active env location : /Users/kelvin/anaconda3/envs/isis
            shell level : 2
       user config file : /Users/kelvin/.condarc
 populated config files : /Users/kelvin/.condarc
          conda version : 4.8.2
    conda-build version : 3.18.11
         python version : 3.7.1.final.0
       virtual packages : __osx=10.13.6
       base environment : /Users/kelvin/anaconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/usgs-astrogeology/osx-64
                          https://conda.anaconda.org/usgs-astrogeology/noarch
                          https://conda.anaconda.org/menpo/osx-64
                          https://conda.anaconda.org/menpo/noarch
                          https://conda.anaconda.org/anaconda/osx-64
                          https://conda.anaconda.org/anaconda/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/kelvin/anaconda3/pkgs
                          /Users/kelvin/.conda/pkgs
       envs directories : /usgs/cpkgs/anaconda3_macOS/envs
                          /Users/kelvin/anaconda3/envs
                          /Users/kelvin/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.2 requests/2.21.0 CPython/3.7.1 Darwin/17.7.0 OSX/10.13.6
                UID:GID : 503:20
             netrc file : None
           offline mode : False

```
</details>

**Comment by jessemapel** [comment]
This is fixable with a patch. An exact version of Python needs to be specified because even with the suggested search directory, cmake still wants to use the higher 3.7 version from the framework.

**Comment by Kelvinrr** [comment]
Assuming the package uploaded days ago has the chances from #22, it still seems to be happening: 

```
igswzawglt0091:isisbuild kelvin$ spiceinit -HELP
dyld: Library not loaded: @rpath/Python3.framework/Versions/3.7/Python3
  Referenced from: /Users/kelvin/anaconda3/envs/neoneoisis/lib/libale.0.dylib
  Reason: no suitable image found.  Did find:
```

Ale install: 

```
# Name                    Version                   Build  Channel
ale                       0.8.0            py37ha1cc60f_1    conda-forge
```

**Comment by jessemapel** [comment]
The osx 3.6 build appears to have found the correct Python:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=148734&view=logs&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=113b264d-9551-5065-da10-7eb3bdeaf1bc&l=177

I'm not sure why this is still happening. I'll look through the rest of the conda-build log later today.

**Comment by jessemapel** [comment]
Seems to be fixed with the 0.8.1 build

```
(ale_py_test) igswzawglt1183:~ jmapel$ otool -L /Users/jmapel/miniconda3/envs/ale_py_test/lib/libale.dylib 
/Users/jmapel/miniconda3/envs/ale_py_test/lib/libale.dylib:
	@rpath/libale.0.dylib (compatibility version 0.0.0, current version 0.8.0)
	@rpath/libpython3.6m.dylib (compatibility version 3.6.0, current version 3.6.0)
	@rpath/libc++.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
```
